istio-control/istio-discovery/values.yaml

```diff
@@ -1,17 +1,68 @@
+# Values for discovery and mesh wide config
 
-# The main components of an environment:
-# - location of discovery, telemetry, flags used in injecting the pod
-# - the discovery environment will also have specific settings
+## Discovery Settings
 
+autoscaleEnabled: true
+autoscaleMin: 1
+autoscaleMax: 5
+replicaCount: 1
+
+# Can be a full hub/image:tag
+image: pilot
+traceSampling: 1.0
+
+# Resources for a small pilot install
+resources:
+  requests:
+    cpu: 500m
+    memory: 2048Mi
+
+env:
+  GODEBUG: gctrace=1
+  CONFIG_NAMESPACE: "istio-config"
+
+cpu:
+  targetAverageUtilization: 80
+
+# Specify the pod anti-affinity that allows you to constrain which nodes
+# your pod is eligible to be scheduled based on labels on pods that are
+# already running on the node rather than based on labels on nodes.
+# There are currently two types of anti-affinity:
+#    "requiredDuringSchedulingIgnoredDuringExecution"
+#    "preferredDuringSchedulingIgnoredDuringExecution"
+# which denote “hard” vs. “soft” requirements, you can define your values
+# in "podAntiAffinityLabelSelector" and "podAntiAffinityTermLabelSelector"
+# correspondingly.
+# For example:
+# podAntiAffinityLabelSelector:
+# - key: security
+#   operator: In
+#   values: S1,S2
+#   topologyKey: "kubernetes.io/hostname"
+# This pod anti-affinity rule says that the pod requires not to be scheduled
+# onto a node if that node is already running a pod with label having key
+# “security” and value “S1”.
+podAntiAffinityLabelSelector: {}
+podAntiAffinityTermLabelSelector: {}
+
+# The following is used to limit how long a sidecar can be connected
+# to a pilot. It balances out load across pilot instances at the cost of
+# increasing system churn.
+keepaliveMaxServerConnectionAge: 30m
+
+# Additional labels to apply to the deployment.
+# The settings will auto-update pilot if keel is installed.
+deploymentLabels:
```

Comment:
uhh, what is keel?

Reply:
The diff is bad here, but I didn't add these. Still a valid question though @costinm any idea why this was added?

Reply:
Keel is an app that auto-updates deployments when new images are pushed ( based on semver) - very useful in some cases.

Reply:
Seems like a perfect use for kustomize and not something we should have in our templates?